From bb7ddc3616a62390195cbaf5b674cb9afab2ad4f Mon Sep 17 00:00:00 2001
From: Tom White <tom@cloudera.com>
Date: Wed, 28 Dec 2011 10:50:26 -0800
Subject: [PATCH 032/795] MR1: CLOUDERA-BUILD. Support TaskID.getTaskType().
 (cherry picked from commit 284575b673cbf7951748b2a74f07a984470ba1bd)

(cherry picked from commit 387a5da7b0c1a1d17002f06e286090a55439e61d)
---
 .../org/apache/hadoop/mapreduce/TaskAttemptID.java |    5 +++++
 src/mapred/org/apache/hadoop/mapreduce/TaskID.java |    7 +++++++
 2 files changed, 12 insertions(+), 0 deletions(-)

diff --git a/src/mapred/org/apache/hadoop/mapreduce/TaskAttemptID.java b/src/mapred/org/apache/hadoop/mapreduce/TaskAttemptID.java
index 6d87289..939a736 100644
--- a/src/mapred/org/apache/hadoop/mapreduce/TaskAttemptID.java
+++ b/src/mapred/org/apache/hadoop/mapreduce/TaskAttemptID.java
@@ -91,6 +91,11 @@ public boolean isMap() {
     return taskId.isMap();
   }
   
+  /**Returns the TaskType of the TaskAttemptID */
+  public TaskType getTaskType() {
+    return taskId.getTaskType();
+  }
+  
   @Override
   public boolean equals(Object o) {
     if (!super.equals(o))
diff --git a/src/mapred/org/apache/hadoop/mapreduce/TaskID.java b/src/mapred/org/apache/hadoop/mapreduce/TaskID.java
index 60377ff..a3bbc46 100644
--- a/src/mapred/org/apache/hadoop/mapreduce/TaskID.java
+++ b/src/mapred/org/apache/hadoop/mapreduce/TaskID.java
@@ -96,6 +96,13 @@ public boolean isMap() {
     return isMap;
   }
   
+  /**
+   * Get the type of the task
+   */
+  public TaskType getTaskType() {
+    return isMap ? TaskType.MAP : TaskType.REDUCE;
+  }
+  
   @Override
   public boolean equals(Object o) {
     if (!super.equals(o))
-- 
1.7.0.4

